File: STRINGS/1768/main.rb
# frozen_string_literal:true

def print_tree(n)
  whitespaces = n / 2
  treesize = 1
  while treesize <= n
    puts ' ' * whitespaces + '*' * treesize
    whitespaces -= 1
    treesize += 2
  end
  whitespaces = n / 2
  puts ' ' * whitespaces + '*'
  puts ' ' * (whitespaces - 1) + '***'
  puts
end

while line = gets
  n = line.to_i
  print_tree(n)
end

File: STRINGS/1768/main.js
function printTree(n) {
    let whitespaces = parseInt(n / 2);
    let treesize = 1;
    while (treesize <= n) {
        console.log(' '.repeat(whitespaces) + '*'.repeat(treesize));
        whitespaces--;
        treesize += 2;
    }
    whitespaces = parseInt(n / 2);
    console.log(' '.repeat(whitespaces) + '*');
    console.log(' '.repeat(--whitespaces) + '***');
    console.log();
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');
let line;
while (line = lines.shift()) {
    let n = parseInt(line);
    printTree(n);
}
